tests/unit/services/test_project.py

```diff
@@ -327,47 +424,59 @@ def test_render_for_platform_independent(
 @pytest.mark.parametrize("platform", ["invalid"])
 @pytest.mark.usefixtures("fake_project_file")
 def test_render_for_invalid_platform(
-    project_service: ProjectService, build_for, build_on, platform
+    real_project_service: ProjectService, build_for, build_on, platform
 ):
     with pytest.raises(errors.InvalidPlatformError) as exc_info:
-        project_service.render_for(
+        real_project_service.render_for(
             build_for=build_for, build_on=build_on, platform=platform
         )
 
     assert cast(str, exc_info.value.details).startswith("Valid platforms are: '")
 
 
+@pytest.mark.parametrize(
+    "build_for", [*(arch.value for arch in craft_platforms.DebianArchitecture), "all"]
+)
+@pytest.mark.usefixtures("fake_project_file")
+def test_cannot_reconfigure(
+    real_project_service: ProjectService, build_for, fake_platform
+):
+    real_project_service.configure(platform=fake_platform, build_for=build_for)
+
+    # Test that we can't re-render no matter the arguments.
```

Comment:
questions: Why is it necessary to test for every combination here? Is it because you expect at some point reconfiguring will become possible for a given combination?

suggestion: if we still want to do that it should probably be a parametrized test.

Reply:
I'm going to guess that I wrote this really late at night and thought I was being thorough. Realistically we just need to check that you can't configure twice.